### PR TITLE
Add a root object for serialization in the Microstream rest api

### LINKED ISSUE #33: Microstream Rest - Root Object
### Issue description

Create a POJO which matches this description: 

https://github.com/microstream-one/microstream/blob/master/storage/rest/service-sparkjava/src/main/resources/doc.json#L12-L21

see example of tests: https://github.com/micronaut-projects/micronaut-chatbots/blob/master/chatbots-telegram-http-client/src/test/groovy/io/micronaut/chatbots/telegram/core/AnimationSpec.groovy

### PR DESCRIPTION
**Pull request by timyates**
Fixes #33

### CODE REVIEW

File: microstream-rest/src/main/java/io/micronaut/microstream/rest/RootObject.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+@Serdeable
+public class RootObject {
+
+    /**
+     * root name.
+     */
+    @NonNull
+    @NotBlank
+    private String name;
```
Comment (sdelamo) [inline microstream-rest/src/main/java/io/micronaut/microstream/rest/RootObject.java:38]: can we make the fields final, add a constructor and remove the setters?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: microstream-rest/src/main/java/io/micronaut/microstream/rest/RootObject.java (hunk 1 above)
Comment (timyates) [reply microstream-rest/src/main/java/io/micronaut/microstream/rest/RootObject.java]: 👍

**Comment by timyates** [comment]
Must be merged after #36

**Review by sdelamo — APPROVED** [review]

**Comment by timyates** [comment]
I can't merge this, as the CLA hasn't passed...  The cla assistant site seems unresponsive 😔

**Pull request merged**